File: packages/python/src/tbd_agents/resources/tokens.py
from __future__ import annotations

from typing import Any, Mapping

from tbd_agents.types import Token, TokenCreate, TokenUpdate

from .base import BaseResource


class TokensResource(BaseResource):
    def create(self, payload: TokenCreate | Mapping[str, Any]) -> Token:
        return self._post("/tokens", payload, model=Token)

    def list(self) -> list[Token]:
        return self._get_list("/tokens", model=Token)

    def get(self, token_id: str) -> Token:
        return self._get(f"/tokens/{token_id}", model=Token)

    def update(self, token_id: str, payload: TokenUpdate | Mapping[str, Any]) -> Token:
        return self._put(f"/tokens/{token_id}", payload, model=Token)

    def delete(self, token_id: str) -> None:
        self._delete(f"/tokens/{token_id}")

File: packages/flutter/lib/src/resources/tokens.dart
/// Tokens resource — `/api/tokens`.
library;

import 'base.dart';

class TokensResource extends BaseResource {
  const TokensResource(super.client);

  Future<List<Map<String, dynamic>>> list() async {
    final data = await client.request('GET', 'tokens');
    if (data is List) return data.whereType<Map<String, dynamic>>().toList();
    return const [];
  }

  Future<Map<String, dynamic>> get(String id) async {
    final data = await client.request('GET', 'tokens/$id');
    return (data as Map<String, dynamic>?) ?? {};
  }

  Future<Map<String, dynamic>> create(Map<String, dynamic> payload) async {
    final data = await client.request('POST', 'tokens', body: payload);
    return (data as Map<String, dynamic>?) ?? {};
  }

  Future<Map<String, dynamic>> update(
      String id, Map<String, dynamic> payload) async {
    final data = await client.request('PUT', 'tokens/$id', body: payload);
    return (data as Map<String, dynamic>?) ?? {};
  }

  Future<void> delete(String id) =>
      client.request('DELETE', 'tokens/$id');
}
